caffe2/core/operator.h

```diff
@@ -120,11 +151,19 @@ class CAFFE2_API OperatorBase : public Observable<OperatorBase> {
   // TODO(jerryzh): Remove this template
   template <typename T>
   inline T* Output(int idx, DeviceType type) {
-    static_assert(
-        std::is_same<T, Tensor>::value,
-        "Output(int, DeviceType) is only available for Tensor");
-    // When you get a Tensor here it is not fully initialized
-    return BlobGetMutableTensor(outputs_.at(idx), type);
+    if (isLegacyOperator()) {
+      static_assert(
+          std::is_same<T, Tensor>::value,
+          "Output(int, DeviceType) is only available for Tensor");
+      // When you get a Tensor here it is not fully initialized
+      return BlobGetMutableTensor(outputs_.at(idx), type);
+    }
+    auto& ival = ivalue_outputs_[idx];
```

Comment:
this is not sufficient:
1) you assume the output is already tensor - what  if it's not?
2) this is not the api we'd be calling - instead majority of places call `OutputTensor(int idx, at::IntList dims, at::TensorOptions options)` below which avoids reinitialization/dtype change in-place and instead creates new tensor input

in both cases we'd need to change ivalue in-place, i.e. we need a pointer to ivalue for this case

Reply:
makes sense -- updated